apps/discussions/app/App.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react'
+import { useAragonApi } from '@aragon/api-react'
+import { Main, Button, TextInput } from '@aragon/ui'
+import styled from 'styled-components'
+
+import { ipfs } from './ipfs'
+
+function App() {
+  const { api, appState, connectedAccount } = useAragonApi()
+  const { syncing } = appState
+
+  const [text, setText] = useState('')
+  const [id, setId] = useState(-1)
+
+  const post = async () => {
+    const discussionPost = {
+      author: connectedAccount,
+      mentions: [],
+      type: 'Post',
+      text,
+    }
+
+    try {
+      if (id === -1) throw new Error('must enter iD')
+      const result = await ipfs.dag.put(discussionPost, {})
+      const cid = result.toBaseEncodedString()
+      await api.post(cid, id.toString()).toPromise()
+      setText('')
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const revise = async () => {
+    let discussionPost = appState.discussions['123'][id]
+    discussionPost.text = text
+    discussionPost.revisions.push(discussionPost.postCid)
+    delete discussionPost.postCid
+    try {
+      const result = await ipfs.dag.put(discussionPost, {})
+      const cid = result.toBaseEncodedString()
+      await api.revise(cid, id, '123').toPromise()
+      setText('')
+      setId(-1)
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const hide = async () => {
+    try {
+      await api.hide(id, '123').toPromise()
+      setId(-1)
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  console.log(appState)
+  return (
+    <Main>
+      <BaseLayout>
+        {syncing && <Syncing />}
+        <TextInput
+          placeholder="write a discussion"
+          value={text}
+          onChange={e => setText(e.target.value)}
+        />
+        Enter post Id to hide or revise post
+        <TextInput.Number
+          placeholder="enter post ID to hide"
+          value={id}
+          onChange={e => setId(Number(e.target.value))}
+        />
+        <Buttons>
+          <Button mode="secondary" onClick={post}>
+            Make post
+          </Button>
+          <Button mode="secondary" onClick={revise}>
+            Revise post
+          </Button>
+          <Button mode="secondary" onClick={hide}>
+            Hide post
+          </Button>
+          <Button
+            mode="secondary"
+            onClick={async () => {
+              const events = await api.pastEvents('0', 'latest').toPromise()
+              console.log(events, 'VENTS')
```

Comment:
@Schwartz10 what's this button for? I'd like to remove this `console.log` – can I remove the whole button?